FJAM-Studios/pixtery
# Cloud functions

This has:
- cloud functions set up (in "functions" folder which is why there are a lot of files changed / added)
- converted uploadPuzzleSettings (in Home) and queryPuzzle (in App) to cloud functions, given that these functions currently expose the database collection name on the client.
- As discussed with Jeff, another function we might want to convert is the one that downloads the URL, given that it is likely giving us cross origin issues when used with web version, but figures it makes sense to wait until the web files are merged so that that can be tested (and also to keep this PR manageable)

**To test:**
At the moment, I am using Firebase Cloud Functions emulator to test and run the cloud functions. Given we need to upgrade to a paid plan to deploy the cloud functions, my sense is that it is probably better to wait to deploy until the cloud functions are more fleshed out. 

Steps to test:
- Set up emulator: emulator set up docs are [here](https://firebase.google.com/docs/emulator-suite/install_and_configure) but I think the steps to get the emulator going are: 
1) download Java https://www.java.com/en/
2) put in your expo LAN IP address that is displayed in Metro bundler where indicated in FirebaseApp file

- run emulator: cd into functions folder, and 'npm run serve'
- expo start in the main pixtery folder (will need separate terminal)
